lib/socket_proxy/forwarder.ex

```diff
@@ -8,18 +8,31 @@ defmodule SocketProxy.Forwarder do
 
   def init({host, port}) do
     send(self(), :connect)
-    {:ok, %{host: host, port: port, socket: nil}}
+    {:ok, %{host: host, port: port, socket: nil, sent: 0}}
   end
 
   def handle_info(:connect, state) do
+    # always close the socket, zero it out in the state, before connecting
+    # to prevent a socket leak
+    if state.socket do
+      :ok = :gen_tcp.close(state.socket)
+    end
+
+    state = %{state | socket: nil, sent: 0}
+
     case :gen_tcp.connect(
            state.host,
            state.port,
            [:binary, active: true, send_timeout: 5_000],
            3_000
          ) do
       {:ok, socket} ->
-        Logger.info("SocketProxy.Forwarder connected to socket #{Util.format_socket(socket)}")
+        Logger.info(
+          "SocketProxy.Forwarder connected to socket=#{Util.format_socket(socket)} port=#{
+            inspect(socket)
```

Comment:
So when I look through the Splunk logs I see this producing output like `port=#Port<0.565>`, do we know if that `565` is actually a TCP port or is it some internal OTP identifier that doesn't mean anything?

Reply:
It's an [erlang port](http://erlang.org/doc/tutorial/c_port.html), i.e. a concept related to interop with the BEAM and non-erlang code.

Reply:
So it doesn't mean anything related to TCP. I included it because if the `socket` gets closed, then `Util.format_socket(socket)` prints `{:error, :einval}`, so by logging the `socket` itself (which, by inspection, I saw was represented by its erlang port), I thought it would allow tying the log back to when it was opened and what IP addresses and TCP ports it corresponded to.

Reply:
Ah, okay, makes sense. 👍